lib/iaas/ComputeClient.js

```diff
@@ -1,22 +1,34 @@
 'use strict';
 
+const _ = require('lodash');
 const Promise = require('bluebird');
 const AWS = require('aws-sdk');
 const MsRestAzure = require('ms-rest-azure');
 const ComputeManagementClient = require('azure-arm-compute');
 const errors = require('../errors');
+const NotImplemented = errors.NotImplemented;
 
 class ComputeClient {
   static createComputeClient(settings) {
     return Promise
       .try(() => {
         switch (settings.provider) {
         case 'amazon':
-          return new AWS.EC2({
-            accessKeyId: settings.keyId,
-            secretAccessKey: settings.key,
-            region: settings.region
-          });
+          {
+            let ec2Config = {
+              accessKeyId: settings.keyId,
+              secretAccessKey: settings.key,
+              region: settings.region,
+              maxRetries: 10
+            };
+            if (Number.isInteger(settings.max_retries) &&
```

Comment:
unnecessary check for Number. Nowhere in our code do we check for the validity of the data type in settings.yml. Whatever is the value in the settings.yml its assumed its set with right data type (for ex, so many durations in milliseconds). If you start doing such validations only at one place it seems to break the consistency. I would recommend removing this.

 So if you remove this explicit numeric check you can set the initial value directly to ec2Config variable this way:
```
 ...
 maxRetries: settings.max_retries || CONST.AWS_MAX_RETRIES
}
```

Reply:
what if someone assigns string value ?